### PR TITLE
Add MITRE ATT&CK Flow form and tools

### PR DESCRIPTION
**Pull request by invisig0th**
feat: `it:mitre:attack:flow` form
feat: `$lib.infosec.mitre.attack.flow` storm library
feat: Implemented MITRE ATT&CK Flow diagram validation and parsing
feat: Updated jsonschema validation code to support handling refs from local disk

Resolves: SYN-4807, SYN-4808

### CODE REVIEW

File: synapse/data/__init__.py [hunk 1]
```diff
@@ -17,8 +25,61 @@ def get(name, defval=None):
...
+    with s_datfile.openDatFile(f'synapse.data/{name}.json') as fd:
+        return json.loads(fd.read())
+
 def path(*names):
     return s_common.genpath(dirname, *names)
+
+def refHandler(func):
+    '''
+    Simple decorator for jsonschema ref handlers to allow for an automatic
+    default behavior of fetching the schema if the custom ref handlers returns
```
Comment (invisig0th) [inline synapse/data/__init__.py:41]: blow up rather than fall back

File: synapse/lib/config.py [hunk 2]
```diff
@@ -49,7 +49,7 @@ def getJsSchema(confbase, confdefs):
     props.update(confbase)
     return schema
 
-def getJsValidator(schema, use_default=True):
+def getJsValidator(schema, use_default=True, handlers=None):
```
Comment (invisig0th) [inline synapse/lib/config.py:52]: no handlers option, just do it :)

File: synapse/lib/stormlib/infosec.py [hunk 3]
```diff
@@ -409,6 +409,60 @@ def CVSS3_1_calc(vdict):
...
+                }
+                $objs_bytype.$type.append($obj)
+            }
+
+            $attack_flow = $objs_bytype."attack-flow".0
+            $created_by = $objs_byid.($attack_flow.created_by_ref)
+
+            [ it:mitre:attack:flow = $guid
+                :name ?= $attack_flow.name
+                :json ?= $flow
```
Comment (invisig0th) [inline synapse/lib/stormlib/infosec.py:452]: not conditional anymore

File: synapse/lib/stormlib/infosec.py [hunk 4]
```diff
@@ -409,6 +409,60 @@ def CVSS3_1_calc(vdict):
...
+                :created ?= $attack_flow.created
+                :updated ?= $attack_flow.modified
+                :author:user ?= $lib.user.iden
+                :author:contact ?= {[ ps:contact = (attack-flow, $created_by.name, $created_by.contact_information)
```
Comment (invisig0th) [inline synapse/lib/stormlib/infosec.py:456]: not conditional either

Comment (invisig0th) [reply synapse/lib/stormlib/infosec.py:456]: trycast the email and name *before* using in the guid grinder

File: synapse/lib/stormtypes.py [hunk 5]
```diff
@@ -1458,8 +1458,14 @@ async def _cast(self, name, valu):
 
         typeitem = self.runt.snap.core.model.type(name)
         if typeitem is None:
-            mesg = f'No type found for name {name}.'
-            raise s_exc.NoSuchType(mesg=mesg)
+            # If a type cannot be found for the form, see if name is a property
+            # that has a type we can use
+            typeitem = self.runt.snap.core.model.prop(name)
```
Comment (invisig0th) [inline synapse/lib/stormtypes.py:1463]: prop =

File: synapse/lib/stormtypes.py [hunk 6]
```diff
@@ -1473,8 +1479,14 @@ async def trycast(self, name, valu):
...
```
Comment (invisig0th) [inline synapse/lib/stormtypes.py:1484]: prop =

**Comment by MichaelSquires** [comment]
Closed in favor of #3361.

**Pull request closed without merging**